### PR TITLE
[Bug][Storage] fix default value get wrong result because no implement read_by_rowids

### LINKED ISSUE #10563: [Bug] select default value error when enable_vectorized_engine = true
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

CREATE TABLE schema_change_update_regression_test (
                `user_id` LARGEINT NOT NULL COMMENT "用户id",
                `date` DATE NOT NULL COMMENT "数据灌入日期时间",
                `city` VARCHAR(20) COMMENT "用户所在城市",
                `age` SMALLINT COMMENT "用户年龄",
                `sex` TINYINT COMMENT "用户性别",
                `last_visit_date` DATETIME DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次访问时间",
                `last_update_date` DATETIME DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次更新时间",
                `last_visit_date_not_null` DATETIME NOT NULL DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次访问时间",
                `cost` BIGINT DEFAULT "0" COMMENT "用户总消费",
                `max_dwell_time` INT DEFAULT "0" COMMENT "用户最大停留时间",
                `min_dwell_time` INT DEFAULT "99999" COMMENT "用户最小停留时间")
            UNIQUE KEY(`user_id`, `date`, `city`, `age`, `sex`) DISTRIBUTED BY HASH(`user_id`)
            PROPERTIES ( "replication_num" = "1" );

INSERT INTO schema_change_update_regression_test VALUES
             (1, '2017-10-01', 'Beijing', 10, 1, '2020-01-01', '2020-01-01', '2020-01-01', 1, 30, 20);

INSERT INTO schema_change_update_regression_test VALUES
             (2, '2017-10-01', 'Beijing', 10, 1, '2020-01-02', '2020-01-02', '2020-01-02', 1, 31, 21);

ALTER table schema_change_update_regression_test ADD COLUMN new_column INT default "1";

SELECT * FROM schema_change_update_regression_test order by user_id DESC, last_visit_date;

+---------+------------+---------+------+------+---------------------+---------------------+--------------------------+------+----------------+----------------+------------+
| user_id | date       | city    | age  | sex  | last_visit_date     | last_update_date    | last_visit_date_not_null | cost | max_dwell_time | min_dwell_time | new_column |
+---------+------------+---------+------+------+---------------------+---------------------+--------------------------+------+----------------+----------------+------------+
| 2       | 2017-10-01 | Beijing |   10 |    1 | 2020-01-02 00:00:00 | 2020-01-02 00:00:00 | 2020-01-02 00:00:00      |    1 |             31 |             21 |          1 |
| 1       | 2017-10-01 | Beijing |   10 |    1 | 2020-01-01 00:00:00 | 2020-01-01 00:00:00 | 2020-01-01 00:00:00      |    1 |             30 |             20 |          0 |
+---------+------------+---------+------+------+---------------------+---------------------+--------------------------+------+----------------+----------------+------------+

new_column should be '1'.

### What You Expected?

default value is correct.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #10563

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 1]
```diff
@@ -990,15 +992,23 @@ void DefaultValueColumnIterator::insert_default_data(const TypeInfo* type_info,
...
+                                                  vectorized::MutableColumnPtr& dst) {
+    _insert_many_default(dst, count);
+    return Status::OK();
+}
+
+void DefaultValueColumnIterator::_insert_many_default(vectorized::MutableColumnPtr& dst, size_t n) {
     if (_is_default_value_null) {
-        *has_null = true;
-        dst->insert_many_defaults(*n);
+        dst->insert_many_defaults(n);
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:1008]: has_null is not set?

Comment (BiteTheDDDDt) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:1008]: set it at line 995

**Review by mrhhsg — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/column_reader.cpp (hunk 1 above)
Comment (yiguolei) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:1008]: read_by_rowids also use _insert_many_default, should it be set?

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**